webscraper.js

```diff
@@ -5,76 +5,64 @@ var driver = new webdriver.Builder().
 withCapabilities(webdriver.Capabilities.safari()).
 build();
 
+function writeToFile(path, thing){
+  writeFile(path, thing, 'utf-8', (err) => {
+    if (err) {
+      console.log("ERROR: ", err);
+      throw err;
+    }
+    console.log(path, "DONE");
+  });
+}
+
+function collectCSS(dir) {
+  let callback = arguments[arguments.length - 1];
+  let node_properties = {};
+  const src = document.documentElement.outerHTML;
+  const all = document.getElementsByTagName("*");
+  for (let j = 0; j < all.length; j++) {
+       const curr = all[j].getBoundingClientRect();
```

Comment:
You might factor out `all[j]` so it doesn't have to compute it (and we don't have to read it) more than once.